### ISSUE TITLE
Report findings as Security Hotspots (instead of violations)

### ISSUE THREAD

**Issue by alixwar** [dependency-check/dependency-check-sonar-plugin#249]
**Is your feature request related to a problem? Please describe.**
OWASP dependency check will find vulnerabilities in dependencies which may not be a problem in practice (for example when the library is not used in a way that would make it vulnerable).

**Describe the solution you'd like**
I think all findings should be reported as security hotspots (https://docs.sonarqube.org/latest/user-guide/security-hotspots/) and then the developer can review the findings and confirm violations if this is the case.

**Describe alternatives you've considered**
There are no alternatives. The current implementation reports everything as violations directly causing annoying false positives.

**Comment by Reamer** [comment]
Hi @alixwar,
This plugin tries to link all SonarQube issues to the correct line in your project configuration file. Only SonarQube issues associated with files are unique and will not be reported again on the next SonarQube run if they are marked as solved. Linking is quite difficult because we don't find any transitive dependencies in `build.gradle` or` pom.xml`. If we don't find the right line, the SonarQube issue is linked to the first line.

Because of this situation, I would prefer a new configuration option to mark problems as a security hotspot (default: disabled).

_Security-Hotspot_ is a Rule-Flag. The change would look like this (Pseudo Code)

```
        rule.setSeverity(Severity.MAJOR);
        rule.setStatus(RuleStatus.READY);
        if (config.markSecurityHotsport) {
            rule.setType(RuleType.SECURITY_HOTSPOT);
         }
        rule.addOwaspTop10(OwaspTop10.A9);
        rule.addCwe(CWE_937);
```
However, this activates the security hotspot for all SonarQube issues created by this plugin, including minor or information issues. Is that workable?

I would advise you and your team to use a suppression file for false positives. Your review will then take place in your SCM. This solution works without plugin code changes.

Please decide Security-Hotspot to all Issues or a suppression-file.

**Comment by alixwar** [comment]
Hi @Reamer ,
thanks for the suggestions. Suppression files could be a viable workaround but that would be to accept false positive violations temporarily which I somehow disapprove of. This would mean that build pipelines would fail due to OWASP dependency check findings, forcing you to urgently fix the suppression file which disturbes the workflow.

I don't mind if all findings (even minor, info) are reported as security hotspots really.
The idea with security hotspots is that it requires someone to manually look through the finding and approve them as violations which I think is exactly what people would want to do in this case.

However, note that this feature was introduced in SonarQube 8.0 so prior versions would need to work as today.

**Comment by Reamer** [comment]
> However, note that this feature was introduced in SonarQube 8.0 so prior versions would need to work as today.

I think we have luck. The Security-Hotspot flag was introduced as an API option in [7.5](https://github.com/SonarSource/sonarqube/commit/d42307dd484ab82daecddb4e98b2085ba711b34f)

Unfortunately the pseudo code is wrong and doesn't work at all. It's not possible with to create a rule based on a configuration option. (this makes sense)

I'll add a second rule with Security-Hotspot flag and decide in SonarQube analysis, which rule should be used.

**Comment by alixwar** [comment]
@Reamer Even if I stumbled on this issue that I don't know if it was intermittent or not I am very eager to use the new functionality provided by this PR. Do you have a plan for releasing it?

**Comment by Reamer** [comment]
Hi @alixwar 
I'm waiting for @bigstinky86 or another reviewer for PR #257 that has fixed issue #247. After that I want to update the year in the copyright part and after that, I'll release a new version of this plugin.

**Comment by alixwar** [comment]
@Reamer Thanks for the update
